libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java

```diff
@@ -12,10 +15,21 @@
   private String drivingSide;
 
   public InstructionModel(DistanceFormatter distanceFormatter, RouteProgress progress) {
-    this.progress = progress;
-    double distanceRemaining = progress.currentLegProgress().currentStepProgress().distanceRemaining();
-    stepDistanceRemaining = distanceFormatter.formatDistance(distanceRemaining);
-    this.drivingSide = progress.currentLegProgress().currentStepProgress().step().drivingSide();
+    if (progress != null && distanceFormatter != null) {
```

Comment:
Could we simplify this `if` nested chain somehow? I'm wondering why we didn't run into this 💥 before in the legacy 🤔 What did it change recently?

Reply:
In the core SDK, RouteProgress and every object inside it @Nullable. This was not the case earlier. Plus we are trying to start navigation without a route which was not the case earlier.